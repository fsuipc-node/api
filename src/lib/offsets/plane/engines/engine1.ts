import { Type } from 'fsuipc';

import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';
import { OffsetCategory } from '@shared/offset-category';

export const engine1: OffsetList = {
  engine1ThrottleLever: new Offset({
    value: 0x88C,
    name: 'engine1ThrottleLever',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 throttle lever - percent - negative = reverse',
    convert: 'Math.round({VAL} < 0 ? {VAL} / 4096 * 100 : {VAL} / 16384 * 100)',
    type: Type.Int16,
    permission: 'r',
  }),
  engine1PropLever: new Offset({
    value: 0x88E,
    name: 'engine1PropLever',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 prop lever - percent - negative = reverse',
    convert: 'Math.round({VAL} < 0 ? {VAL} / 4096 * 100 : {VAL} / 16384 * 100)',
    type: Type.Int16,
    permission: 'rw',
  }),
  engine1MixtureLever: new Offset({
    value: 0x890,
    name: 'engine1MixtureLever',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 prop lever - percent',
    convert: '{VAL} / 16384 * 100',
    type: Type.Int16,
    permission: 'rw',
  }),
  engine1StarterSwitchPosition: new Offset({
    value: 0x892,
    name: 'engine1StarterSwitchPosition',
    category: OffsetCategory.ENGINE,
    description: 'engine1 start switch position - JET: 0=off 1=start 2=gen/alt - PROP: 0=off 1=right 2=left 3=both 4=start',
    type: Type.UInt16,
    permission: 'rw',
  }),
  engine1Firing: new Offset({
    value: 0x894,
    name: 'engine1Firing',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 combustion',
    convert: '!!{VAL}',
    type: Type.UInt16,
    permission: 'r',
  }),
  engine1N2: new Offset({
    value: 0x896,
    name: 'engine1N2',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 N2 - also helo RPM %',
    convert: '{VAL} / 16384 * 100',
    type: Type.Int16,
    permission: 'rw',
  }),
  engine1N1: new Offset({
    value: 0x898,
    name: 'engine1N1',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 N1 - also helo RPM %',
    convert: '{VAL} / 16384 * 100',
    type: Type.Int16,
    permission: 'rw',
  }),
  engine1PropRPM: new Offset({
    value: 0x898,
    name: 'engine1PropRPM',
    category: OffsetCategory.ENGINE,
    description: 'derive RPM by multiplying by  - negative = counter-rotating propeller',
    convert: '{VAL} / 16384 / 65536',
    type: Type.Int16,
    permission: 'rw',
  }),
  engine1ThrottleLeverControl: new Offset({
    value: 0x89A,
    name: 'engine1ThrottleLeverControl',
    category: OffsetCategory.ENGINE,
    description: 'engine1 throttle lever control - -4096 to +16384',
    type: Type.Int16,
    permission: 'rw',
  }),
  engine1FuelFlowLbHourSSL: new Offset({
    value: 0x8A0,
    name: 'engine1FuelFlowLbHourSSL',
    category: OffsetCategory.ENGINE,
    description: 'engine1 fuel flow (lb per hour, sea level)',
    convert: '{VAL} / 128',
    type: Type.UInt16,
    permission: 'r',
  }),
  engine1AntiIce: new Offset({
    value: 0x8B2,
    name: 'engine1AntiIce',
    category: OffsetCategory.ENGINE,
    description: 'engine1 anti-ice/carb heat active',
    convert: '!!{VAL}',
    type: Type.UInt16,
    permission: 'r',
  }),
  engine1OilTemp: new Offset({
    value: 0x8B8,
    name: 'engine1OilTemp',
    category: OffsetCategory.ENGINE,
    description: 'engine1 oil temp - celsius',
    convert: 'Math.round({VAL} * 140 / 16384)',
    type: Type.Int16,
    permission: 'r',
  }),
  engine1OilPres: new Offset({
    value: 0x8BA,
    name: 'engine1OilPres',
    category: OffsetCategory.ENGINE,
    description: 'engine1 oil pressure - PSI',
    convert: 'Math.round({VAL} * 55 / 16384)',
    type: Type.UInt16,
    permission: 'r',
  }),
  engine1PressureRatio: new Offset({
    value: 0x8BC,
    name: 'engine1PressureRatio',
    category: OffsetCategory.ENGINE,
    description: 'engine1 pressure ratio',
    convert: '{VAL} * 1.6 / 16384',
    type: Type.Int16,
    permission: 'r',
  }),
  engine1EGT: new Offset({
    value: 0x8BE,
    name: 'engine1EGT',
    category: OffsetCategory.ENGINE,
    description: 'engine1 exhaust gas temperature - Trust only on jet engine - celsius',
    convert: 'Math.round({VAL} * 860 / 16384)',
    type: Type.Int16,
    permission: 'r',
  }),
  engine1MP: new Offset({
    value: 0x8C0,
    name: 'engine1MP',
    category: OffsetCategory.ENGINE,
    description: 'engine1 manifold pressure - inHg',
    convert: '{VAL} / 1024',
    type: Type.UInt16,
    permission: 'r',
  }),
  engine1RPMScaler: new Offset({
    value: 0x8C8,
    name: 'engine1RPMScaler',
    category: OffsetCategory.ENGINE,
    description: 'engine1 RPM Scaler - props: used to calculate RPM',
    type: Type.Int16,
    permission: 'r',
  }),
  engine1OilQuantity: new Offset({
    value: 0x8D0,
    name: 'engine1OilQuantity',
    category: OffsetCategory.ENGINE,
    description: 'engine1 oil quantity - percent',
    convert: 'Math.round({VAL} / 16384 * 100)',
    type: Type.Int32,
    permission: 'r',
  }),
  engine1Vibration: new Offset({
    value: 0x8D4,
    name: 'engine1Vibration',
    category: OffsetCategory.ENGINE,
    description: 'engine1 vibration',
    convert: 'Math.round({VAL} * 5 / 16384)',
    type: Type.Int32,
    permission: 'r',
  }),
  engine1HydPres: new Offset({
    value: 0x8D8,
    name: 'engine1HydPres',
    category: OffsetCategory.ENGINE,
    description: 'engine1 hydraulic pressure - PSI',
    convert: '{VAL} / 4',
    type: Type.Int32,
    permission: 'r',
  }),
  engine1HydQuantity: new Offset({
    value: 0x8DC,
    name: 'engine1HydQuantity',
    category: OffsetCategory.ENGINE,
    description: 'engine1 hydraulic quantity - PSI',
    convert: '{VAL} / 16384 * 100',
    type: Type.Int32,
    permission: 'r',
  }),
  engine1CHT: new Offset({
    value: 0x8E8,
    name: 'engine1CHT',
    category: OffsetCategory.ENGINE,
    description: 'engine1 cylinder head temperature - F',
    type: Type.Double,
    permission: 'r',
  }),
  engine1ITT: new Offset({
    value: 0x8F0,
    name: 'engine1ITT',
    category: OffsetCategory.ENGINE,
    description: 'engine1 turbine temperature - C',
    convert: '{VAL} / 16384',
    type: Type.Int32,
    permission: 'r',
  }),
  engine1Torque: new Offset({
    value: 0x8F4,
    name: 'engine1Torque',
    category: OffsetCategory.ENGINE,
    description: 'engine1 torque - trust only for helo - percent',
    convert: '{VAL} / 16384 * 100',
    type: Type.Int32,
    permission: 'r',
  }),
  engine1FuelPres: new Offset({
    value: 0x8F8,
    name: 'engine1FuelPres',
    category: OffsetCategory.ENGINE,
    description: 'engine1 fuel pressure - PSI',
    convert: '{VAL} / 144',
    type: Type.Int32,
    permission: 'r',
  }),
  engine1FuelUsedSinceStart: new Offset({
    value: 0x90C,
    name: 'engine1FuelUsedSinceStart',
    category: OffsetCategory.ENGINE,
    description: 'engine1 used fuel since start - lb',
    type: Type.Single,
    permission: 'r',
  }),
  engine1FuelElapsedTime: new Offset({
    value: 0x910,
    name: 'engine1FuelElapsedTime',
    category: OffsetCategory.ENGINE,
    description: 'engine1 fuel elapsed time - h',
    type: Type.Single,
    permission: 'r',
  }),
  engine1FuelFlowLbHour: new Offset({
    value: 0x918,
    name: 'engine1FuelFlowLbHour',
    category: OffsetCategory.ENGINE,
    description: 'engine1 fuel elapsed time - lb/h',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TorqueFtLb: new Offset({
    value: 0x920,
    name: 'engine1TorqueFtLb',
    category: OffsetCategory.ENGINE,
    description: 'engine1 fuel elapsed time - ft lb',
    type: Type.Single,
    permission: 'r',
  }),
  engine1TurbineN1: new Offset({
    value: 0x2000,
    name: 'engine1TurbineN1',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine N1 - percent',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineN2: new Offset({
    value: 0x2008,
    name: 'engine1TurbineN2',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine N2 - percent',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineCorrectedN1: new Offset({
    value: 0x2010,
    name: 'engine1TurbineCorrectedN1',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine corrected N1 - percent',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineCorrectedN2: new Offset({
    value: 0x2018,
    name: 'engine1TurbineCorrectedN2',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine corrected N2 - percent',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineCorrectedFuelFlow: new Offset({
    value: 0x2020,
    name: 'engine1TurbineCorrectedFuelFlow',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine corrected fuel flow - lb/h',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineTorque: new Offset({
    value: 0x2028,
    name: 'engine1TurbineTorque',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine corrected fuel flow - 0.0 to 1.0',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineEPR: new Offset({
    value: 0x2030,
    name: 'engine1TurbineEPR',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine EPR',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineITT: new Offset({
    value: 0x2038,
    name: 'engine1TurbineITT',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine ITT',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineAfterburnerActive: new Offset({
    value: 0x2048,
    name: 'engine1TurbineAfterburnerActive',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine afterburner active',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  engine1TurbineJetThrust: new Offset({
    value: 0x204C,
    name: 'engine1TurbineJetThrust',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine jet thrust',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineTankSelector: new Offset({
    value: 0x2054,
    name: 'engine1TurbineTankSelector',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine tank selector',
    convert: 'fuelTank',
    mapping: true,
    type: Type.UInt32,
    permission: 'r',
  }),
  engine1TurbineUsedTank: new Offset({
    value: 0x2058,
    name: 'engine1TurbineUsedTank',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine used tank',
    type: Type.BitArray,
    length: 11,
    permission: 'r',
  }),
  engine1TurbineAvailableTankCount: new Offset({
    value: 0x205C,
    name: 'engine1TurbineAvailableTankCount',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine available tank count',
    type: Type.UInt32,
    permission: 'r',
  }),
  engine1TurbineFuelFlow: new Offset({
    value: 0x2060,
    name: 'engine1TurbineFuelFlow',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine fuel flow',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineIsFuelAvailable: new Offset({
    value: 0x2068,
    name: 'engine1TurbineIsFuelAvailable',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine is fuel available flag',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  engine1TurbineBleedPressure: new Offset({
    value: 0x206C,
    name: 'engine1TurbineBleedPressure',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine bleed air pressure - lb/sq in',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineReverser: new Offset({
    value: 0x207C,
    name: 'engine1TurbineReverser',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine reverser position - 0.0 - 1.0 lb/sq in',
    convert: 'Math.round(+(+({VAL}).toFixed(4) * 100).toFixed(2))',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineVibration: new Offset({
    value: 0x2084,
    name: 'engine1TurbineVibration',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine vibration',
    type: Type.Double,
    permission: 'r',
  }),
  engine1TurbineIgnition: new Offset({
    value: 0x208C,
    name: 'engine1TurbineIgnition',
    category: OffsetCategory.ENGINE,
    description: 'engine 1 turbine is fuel available flag',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'rw',
  }),
};
