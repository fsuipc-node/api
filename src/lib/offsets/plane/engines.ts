import { Type } from 'fsuipc';

import { OffsetCategory } from '@shared/offset-category';
import { OffsetList } from '@shared/offset-list';
import { Offset } from '@shared/offset';

import { engine1 } from './engines/engine1';
import { engine2 } from './engines/engine2';
import { engine3 } from './engines/engine3';
import { engine4 } from './engines/engine4';

export const engines: OffsetList = {
  engineType: new Offset({
    value: 0x609,
    name: 'engineType',
    category: OffsetCategory.ENGINE,
    description: 'engine type',
    convert: 'engineType',
    mapping: true,
    type: Type.Byte,
    permission: 'r',
  }),
  hasMixtureControl: new Offset({
    value: 0x780,
    name: 'hasMixtureControl',
    category: OffsetCategory.ENGINE,
    description: 'has mixture control',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  hasCarbHeat: new Offset({
    value: 0x784,
    name: 'hasCarbHeat',
    category: OffsetCategory.ENGINE,
    description: 'has carb heat',
    convert: '!!{VAL}',
    type: Type.UInt32,
    permission: 'r',
  }),
  activeEngine: new Offset({
    value: 0x888,
    name: 'activeEngine',
    category: OffsetCategory.ENGINE,
    description: 'active engine pattern',
    type: Type.BitArray,
    length: 1,
    permission: 'rw',
  }),
  ...engine1,
  ...engine2,
  ...engine3,
  ...engine4,
};
