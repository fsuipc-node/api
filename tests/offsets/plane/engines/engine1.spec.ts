import { engine1 as offsets } from '@offsets/plane/engines/engine1';

describe('offset - plane/engines/engine1', () => {
  const offsetsTestCases = [
    { name: 'engine1ThrottleLever', value: -2048, expectedResult: -50 },
    { name: 'engine1ThrottleLever', value: 0, expectedResult: 0 },
    { name: 'engine1ThrottleLever', value: 8192, expectedResult: 50 },
    { name: 'engine1PropLever', value: -2048, expectedResult: -50 },
    { name: 'engine1PropLever', value: 0, expectedResult: 0 },
    { name: 'engine1PropLever', value: 8192, expectedResult: 50 },
    { name: 'engine1MixtureLever', value: 8192, expectedResult: 50 },
    { name: 'engine1Firing', value: 0, expectedResult: false },
    { name: 'engine1Firing', value: 1, expectedResult: true },
    { name: 'engine1N2', value: 8192, expectedResult: 50 },
    { name: 'engine1N1', value: 8192, expectedResult: 50 },
    { name: 'engine1PropRPM', value: 107374182400, expectedResult: 100 },
    { name: 'engine1FuelFlowLbHourSSL', value: 256, expectedResult: 2 },
    { name: 'engine1AntiIce', value: 0, expectedResult: false },
    { name: 'engine1AntiIce', value: 1, expectedResult: true },
    { name: 'engine1OilTemp', value: 10532, expectedResult: 90 },
    { name: 'engine1OilPres', value: 22342, expectedResult: 75 },
    { name: 'engine1PressureRatio', value: 20480, expectedResult: 2 },
    { name: 'engine1EGT', value: 14281, expectedResult: 750 },
    { name: 'engine1MP', value: 20480, expectedResult: 20 },
    { name: 'engine1OilQuantity', value: 8167, expectedResult: 50 },
    { name: 'engine1Vibration', value: 6553, expectedResult: 2 },
    { name: 'engine1HydPres', value: 64, expectedResult: 16 },
    { name: 'engine1HydQuantity', value: 8192, expectedResult: 50 },
    { name: 'engine1ITT', value: 3276800, expectedResult: 200 },
    { name: 'engine1Torque', value: 8192, expectedResult: 50 },
    { name: 'engine1FuelPres', value: 288, expectedResult: 2 },
  ];

  describe('offsets list', () => {
    it('should have required properties', () => {
      expect(offsets).toMatchSnapshot();
    });
  });

  offsetsTestCases.forEach(testedOffset => {
    describe(testedOffset.name, () => {
      it('should convert data properly', () => {
        const convertExpression = offsets[testedOffset.name].convert.replace(new RegExp(/{VAL}/g), testedOffset.value.toString());

        // tslint:disable-next-line:no-eval
        expect(eval(convertExpression)).toEqual(testedOffset.expectedResult);
      });
    });
  });
});
