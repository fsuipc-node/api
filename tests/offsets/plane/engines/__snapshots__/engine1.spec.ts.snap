// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offset - plane/engines/engine1 offsets list should have required properties 1`] = `
Object {
  "engine1AntiIce": Offset {
    "category": "engine",
    "convert": "!!{VAL}",
    "description": "engine1 anti-ice/carb heat active",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1AntiIce",
    "permission": "r",
    "type": 5,
    "value": 2226,
  },
  "engine1CHT": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 cylinder head temperature - F",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1CHT",
    "permission": "r",
    "type": 8,
    "value": 2280,
  },
  "engine1EGT": Offset {
    "category": "engine",
    "convert": "Math.round({VAL} * 860 / 16384)",
    "description": "engine1 exhaust gas temperature - Trust only on jet engine - celsius",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1EGT",
    "permission": "r",
    "type": 2,
    "value": 2238,
  },
  "engine1Firing": Offset {
    "category": "engine",
    "convert": "!!{VAL}",
    "description": "engine 1 combustion",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1Firing",
    "permission": "r",
    "type": 5,
    "value": 2196,
  },
  "engine1FuelElapsedTime": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 fuel elapsed time - h",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1FuelElapsedTime",
    "permission": "r",
    "type": 9,
    "value": 2320,
  },
  "engine1FuelFlowLbHour": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 fuel elapsed time - lb/h",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1FuelFlowLbHour",
    "permission": "r",
    "type": 8,
    "value": 2328,
  },
  "engine1FuelFlowLbHourSSL": Offset {
    "category": "engine",
    "convert": "{VAL} / 128",
    "description": "engine1 fuel flow (lb per hour, sea level)",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1FuelFlowLbHourSSL",
    "permission": "r",
    "type": 5,
    "value": 2208,
  },
  "engine1FuelPres": Offset {
    "category": "engine",
    "convert": "{VAL} / 144",
    "description": "engine1 fuel pressure - PSI",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1FuelPres",
    "permission": "r",
    "type": 3,
    "value": 2296,
  },
  "engine1FuelUsedSinceStart": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 used fuel since start - lb",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1FuelUsedSinceStart",
    "permission": "r",
    "type": 9,
    "value": 2316,
  },
  "engine1HydPres": Offset {
    "category": "engine",
    "convert": "{VAL} / 4",
    "description": "engine1 hydraulic pressure - PSI",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1HydPres",
    "permission": "r",
    "type": 3,
    "value": 2264,
  },
  "engine1HydQuantity": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 * 100",
    "description": "engine1 hydraulic quantity - PSI",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1HydQuantity",
    "permission": "r",
    "type": 3,
    "value": 2268,
  },
  "engine1ITT": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384",
    "description": "engine1 turbine temperature - C",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1ITT",
    "permission": "r",
    "type": 3,
    "value": 2288,
  },
  "engine1MP": Offset {
    "category": "engine",
    "convert": "{VAL} / 1024",
    "description": "engine1 manifold pressure - inHg",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1MP",
    "permission": "r",
    "type": 5,
    "value": 2240,
  },
  "engine1MixtureLever": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 * 100",
    "description": "engine 1 prop lever - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1MixtureLever",
    "permission": "rw",
    "type": 2,
    "value": 2192,
  },
  "engine1N1": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 * 100",
    "description": "engine 1 N1 - also helo RPM %",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1N1",
    "permission": "rw",
    "type": 2,
    "value": 2200,
  },
  "engine1N2": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 * 100",
    "description": "engine 1 N2 - also helo RPM %",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1N2",
    "permission": "rw",
    "type": 2,
    "value": 2198,
  },
  "engine1OilPres": Offset {
    "category": "engine",
    "convert": "Math.round({VAL} * 55 / 16384)",
    "description": "engine1 oil pressure - PSI",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1OilPres",
    "permission": "r",
    "type": 5,
    "value": 2234,
  },
  "engine1OilQuantity": Offset {
    "category": "engine",
    "convert": "Math.round({VAL} / 16384 * 100)",
    "description": "engine1 oil quantity - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1OilQuantity",
    "permission": "r",
    "type": 3,
    "value": 2256,
  },
  "engine1OilTemp": Offset {
    "category": "engine",
    "convert": "Math.round({VAL} * 140 / 16384)",
    "description": "engine1 oil temp - celsius",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1OilTemp",
    "permission": "r",
    "type": 2,
    "value": 2232,
  },
  "engine1PressureRatio": Offset {
    "category": "engine",
    "convert": "{VAL} * 1.6 / 16384",
    "description": "engine1 pressure ratio",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1PressureRatio",
    "permission": "r",
    "type": 2,
    "value": 2236,
  },
  "engine1PropLever": Offset {
    "category": "engine",
    "convert": "Math.round({VAL} < 0 ? {VAL} / 4096 * 100 : {VAL} / 16384 * 100)",
    "description": "engine 1 prop lever - percent - negative = reverse",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1PropLever",
    "permission": "rw",
    "type": 2,
    "value": 2190,
  },
  "engine1PropRPM": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 / 65536",
    "description": "derive RPM by multiplying by  - negative = counter-rotating propeller",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1PropRPM",
    "permission": "rw",
    "type": 2,
    "value": 2200,
  },
  "engine1RPMScaler": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 RPM Scaler - props: used to calculate RPM",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1RPMScaler",
    "permission": "r",
    "type": 2,
    "value": 2248,
  },
  "engine1StarterSwitchPosition": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 start switch position - JET: 0=off 1=start 2=gen/alt - PROP: 0=off 1=right 2=left 3=both 4=start",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1StarterSwitchPosition",
    "permission": "rw",
    "type": 5,
    "value": 2194,
  },
  "engine1ThrottleLever": Offset {
    "category": "engine",
    "convert": "Math.round({VAL} < 0 ? {VAL} / 4096 * 100 : {VAL} / 16384 * 100)",
    "description": "engine 1 throttle lever - percent - negative = reverse",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1ThrottleLever",
    "permission": "r",
    "type": 2,
    "value": 2188,
  },
  "engine1ThrottleLeverControl": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 throttle lever control - -4096 to +16384",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1ThrottleLeverControl",
    "permission": "rw",
    "type": 2,
    "value": 2202,
  },
  "engine1Torque": Offset {
    "category": "engine",
    "convert": "{VAL} / 16384 * 100",
    "description": "engine1 torque - trust only for helo - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1Torque",
    "permission": "r",
    "type": 3,
    "value": 2292,
  },
  "engine1TorqueFtLb": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine1 fuel elapsed time - ft lb",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TorqueFtLb",
    "permission": "r",
    "type": 9,
    "value": 2336,
  },
  "engine1TurbineAfterburnerActive": Offset {
    "category": "engine",
    "convert": "!!{VAL}",
    "description": "engine 1 turbine afterburner active",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineAfterburnerActive",
    "permission": "r",
    "type": 6,
    "value": 8264,
  },
  "engine1TurbineAvailableTankCount": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine available tank count",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineAvailableTankCount",
    "permission": "r",
    "type": 6,
    "value": 8284,
  },
  "engine1TurbineBleedPressure": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine bleed air pressure - lb/sq in",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineBleedPressure",
    "permission": "r",
    "type": 8,
    "value": 8300,
  },
  "engine1TurbineCorrectedFuelFlow": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine corrected fuel flow - lb/h",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineCorrectedFuelFlow",
    "permission": "r",
    "type": 8,
    "value": 8224,
  },
  "engine1TurbineCorrectedN1": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine corrected N1 - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineCorrectedN1",
    "permission": "r",
    "type": 8,
    "value": 8208,
  },
  "engine1TurbineCorrectedN2": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine corrected N2 - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineCorrectedN2",
    "permission": "r",
    "type": 8,
    "value": 8216,
  },
  "engine1TurbineEPR": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine EPR",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineEPR",
    "permission": "r",
    "type": 8,
    "value": 8240,
  },
  "engine1TurbineFuelFlow": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine fuel flow",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineFuelFlow",
    "permission": "r",
    "type": 8,
    "value": 8288,
  },
  "engine1TurbineITT": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine ITT",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineITT",
    "permission": "r",
    "type": 8,
    "value": 8248,
  },
  "engine1TurbineIgnition": Offset {
    "category": "engine",
    "convert": "!!{VAL}",
    "description": "engine 1 turbine is fuel available flag",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineIgnition",
    "permission": "rw",
    "type": 6,
    "value": 8332,
  },
  "engine1TurbineIsFuelAvailable": Offset {
    "category": "engine",
    "convert": "!!{VAL}",
    "description": "engine 1 turbine is fuel available flag",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineIsFuelAvailable",
    "permission": "r",
    "type": 6,
    "value": 8296,
  },
  "engine1TurbineJetThrust": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine jet thrust",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineJetThrust",
    "permission": "r",
    "type": 8,
    "value": 8268,
  },
  "engine1TurbineN1": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine N1 - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineN1",
    "permission": "r",
    "type": 8,
    "value": 8192,
  },
  "engine1TurbineN2": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine N2 - percent",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineN2",
    "permission": "r",
    "type": 8,
    "value": 8200,
  },
  "engine1TurbineReverser": Offset {
    "category": "engine",
    "convert": "Math.round(+(+({VAL}).toFixed(4) * 100).toFixed(2))",
    "description": "engine 1 turbine reverser position - 0.0 - 1.0 lb/sq in",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineReverser",
    "permission": "r",
    "type": 8,
    "value": 8316,
  },
  "engine1TurbineTankSelector": Offset {
    "category": "engine",
    "convert": "fuelTank",
    "description": "engine 1 turbine tank selector",
    "length": undefined,
    "mapping": true,
    "name": "engine1TurbineTankSelector",
    "permission": "r",
    "type": 6,
    "value": 8276,
  },
  "engine1TurbineTorque": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine corrected fuel flow - 0.0 to 1.0",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineTorque",
    "permission": "r",
    "type": 8,
    "value": 8232,
  },
  "engine1TurbineUsedTank": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine used tank",
    "length": 11,
    "mapping": undefined,
    "name": "engine1TurbineUsedTank",
    "permission": "r",
    "type": 12,
    "value": 8280,
  },
  "engine1TurbineVibration": Offset {
    "category": "engine",
    "convert": undefined,
    "description": "engine 1 turbine vibration",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1TurbineVibration",
    "permission": "r",
    "type": 8,
    "value": 8324,
  },
  "engine1Vibration": Offset {
    "category": "engine",
    "convert": "Math.round({VAL} * 5 / 16384)",
    "description": "engine1 vibration",
    "length": undefined,
    "mapping": undefined,
    "name": "engine1Vibration",
    "permission": "r",
    "type": 3,
    "value": 2260,
  },
}
`;
